codegens/csharp-restsharp/lib/util.js

```diff
@@ -1,19 +1,79 @@
 /**
  * sanitizes input string by handling escape characters eg: converts '''' to '\'\'' and trim input if required
- * 
+ *
  * @param {String} inputString - Input String to sanitize
  * @param {Boolean} [trim] - Indicates whether to trim string or not
  * @returns {String} Sanitized String handling escape characters
  */
 function sanitize (inputString, trim) {
     /* istanbul ignore next */
     if (typeof inputString !== 'string') {
-        throw new Error(`Csharp-Restsharp-Converter:function expects input to be string, found ${typeof inputString}`);
+        return '';
     }
     inputString = inputString.replace(/\\/g, '\\\\').replace(/"/g, '\\"');
     return trim ? inputString.trim() : inputString;
 }
 
+/**
+ * sanitizes input options
+ *
+ * @param {Object} options - Options provided by the user
+ * @param {Object} defaultOptions - default options object containing type and default for each property,
+ *  built from getOptions array
+ *
+ * @returns {Object} - Sanitized options object
+ */
+function sanitizeOptions (options, defaultOptions) {
+    var result = {},
+        id;
+
+    for (id in options) {
+        if (options.hasOwnProperty(id)) {
+            if (defaultOptions[id] === undefined) {
+                continue;
+            }
+            switch (defaultOptions[id].type) {
+                case 'boolean':
+                    if (typeof options[id] !== 'boolean') {
+                        result[id] = defaultOptions[id].default;
+                    }
+                    else {
+                        result[id] = options[id];
+                    }
+                    break;
+                case 'integer':
+                    if (typeof options[id] !== 'number' || options[id] < 0) {
+                        result[id] = defaultOptions[id].default;
```

Comment:
@shreys7 This seems wrong. Unless we know what option it is, we can't assume >0.
Can we change the type for indentCount and requestTimeout to `positiveInteger`, and update references?